src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -0,0 +1,1842 @@
+        handle_update_participant_command
+    )
+    app.command(config.slack_command_list_resources, middleware=middleware)(
+        handle_list_resources_command
+    )
+    app.command(config.slack_command_engage_oncall, middleware=middleware)(
+        handle_engage_oncall_command
+    )
+
+    # sensitive commands
+    middleware.append(restricted_command_middleware)
+    app.command(config.slack_command_assign_role, middleware=middleware)(handle_assign_role_command)
+    app.command(config.slack_command_update_incident, middleware=middleware)(
+        handle_update_incident_command
+    )
+    app.command(config.slack_command_update_notifications_group, middleware=middleware)(
+        handle_update_notifications_group_command
+    )
+    app.command(config.slack_command_report_tactical, middleware=middleware)(
+        handle_report_tactical_command
+    )
+    app.command(config.slack_command_report_executive, middleware=middleware)(
+        handle_report_executive_command
+    )
+    app.command(config.slack_command_add_timeline_event, middleware=middleware)(
+        handle_add_timeline_event_command
+    )
+
+    # required because allow the user to change the reaction string
+    app.event(config.timeline_event_reaction, middleware=[db_middleware])(
+        handle_timeline_added_event
+    )
+
+
+def refetch_db_session(organization_slug: str) -> Session:
+    schema_engine = engine.execution_options(
+        schema_translate_map={
+            None: f"dispatch_organization_{organization_slug}",
+        }
+    )
+    db_session = sessionmaker(bind=schema_engine)()
+    return db_session
+
+
+@app.options(
+    DefaultActionIds.tags_multi_select, middleware=[action_context_middleware, db_middleware]
+)
+async def handle_tag_search_action(ack, payload, context, db_session):
+    """Handles tag lookup actions."""
+    query_str = payload["value"]
+
+    filter_spec = {
+        "and": [
+            {"model": "Project", "op": "==", "field": "id", "value": context["subject"].project_id}
+        ]
+    }
+
+    if "/" in query_str:
+        tag_type, query_str = query_str.split("/")
+        filter_spec["and"].append(
+            {"model": "TagType", "op": "==", "field": "name", "value": tag_type}
+        )
+
+    tags = search_filter_sort_paginate(
+        db_session=db_session, model="Tag", query_str=query_str, filter_spec=filter_spec
+    )
+
+    options = []
+    for t in tags["items"]:
+        options.append(
+            {
+                "text": {"type": "plain_text", "text": f"{t.tag_type.name}/{t.name}"},
+                "value": str(
+                    t.id
+                ),  # NOTE slack doesn't not accept int's as values (fails silently)
+            }
+        )
+
+    await ack(options=options)
+
+
+@app.action(
+    IncidentUpdateActions.project_select, middleware=[action_context_middleware, db_middleware]
+)
+async def handle_project_select_action(ack, body, client, context, db_session):
```

Comment:
Shouldn't this function be renamed to `handle_update_incident_project_select_action` to be consistent with other functions?